src/photon_physics.F90

```diff
@@ -305,13 +305,22 @@ recursive subroutine atomic_relaxation(p, elm, i_shell)
     integer :: secondary
     real(8) :: c
     real(8) :: rn
-    real(8) :: E
     real(8) :: mu
     real(8) :: phi
     real(8) :: uvw(3)
+    real(8) :: E
 
-    ! Check for no transitions
-    if (elm % shells(i_shell) % n_transitions == 0) return
+    ! If no transitions, assume fluorescent photon from captured free electron
```

Comment:
The basis for it is just maintaining energy balance.  If an electron hole is created (and especially one from an inner shell electron), the hole is definitely going to be filled.  If there is no shell to shell transition found, we still need to somehow fill it to ensure energy balance.  My thought was just to fill it with a free electron.

I'm not sure what other codes do.  It is in the weeds of handling atomic relaxation so I'm not sure if it will be reported in any theory manuals. I'll check the masters thesis you shared and try to find what is done in the Serpent source code.